elastic/kibana
# [Reporting/Dashboard] Update integration to use v2 reports

## Summary

Closes https://github.com/elastic/kibana/issues/99890

Updates the dashboard-reporting integration so that dashboard uses v2 export types. This enables PNG/PDF reports to be generated without first needing to save as well as using locators instead of storing URLs.


## How to test

1. Start Kibana + ES on basic level license
2. Load sample data into the cluster via the UI
3. Open dashboard and create a new dashboard
4. Add a visulization
5. Add a query to the dashboard
6. Add a filter to the dashboard
7. Set a custom timerange on the dashboard
6. Create a PDF and PNG report via the "Share" menu
7. Copy the POST URL for both the PDF and PNG report and execute these against local Kibana
8. Verify that all the reports have been generated and show the expected result in the visualisations

## To reviewers

* Will the "Copy POST URL" functionality cause any issues? I know URL length is a concern with Vega charts, but looking at the current implementation, is there anything missing?
* I've reworked the locator to pass all state on `scopedHistory.location.state` instead of the URL and created a new interface called `ForwardedDashboardState`. This is the shape of the data dashboard should expect to find on `scopedHistory.location.state`.

## Screenshots

Unsaved dashboard, with unsaved panel (on the left):

<img width="1588" alt="Screenshot 2021-08-19 at 15 29 23" src="https://user-images.githubusercontent.com/8155004/130096137-26ebe553-f69f-412e-aaf6-c2920ed63a02.png">

## Release note

Dashboard PDF and PNG reports can now be generated without requiring users to first save.

![Screenshot 2021-08-17 at 12 27 16](https://user-images.githubusercontent.com/8155004/129710435-ef01fbc1-fb38-4f7b-bdb8-4c1b23f76568.png)



## TODO

- [x] Figure out how to make time filters work correctly
